config/config.go

```diff
@@ -161,6 +166,174 @@ type AuthConfig struct {
 	AppendMetadataHeaders bool          `yaml:"append_metadata_headers,omitempty" json:"append_metadata_headers,omitempty"`
 }
 
+// EnvConfigs
+type EnvConfigs struct {
+	// A list of files containing environment config
+	ConfigFiles []string `yaml:"config_files,omitempty" json:"config_files,omitempty"`
+
+	// A list of environment configs
+	InlineConfigs []EnvironmentConfig `yaml:"inline_configs,omitempty" json:"inline_configs,omitempty"`
+}
+
+// EnvironmentConfig is an Apigee Environment-level config for
+// Envoy Adapter. It contains a list of operations for the adapter to
+// perform request authentication and authorization.
+type EnvironmentConfig struct {
+	// Unique ID of the environment config
+	ID string `yaml:"id" json:"id"`
+
+	// A list of proxy configs
+	ProxyConfigs []ProxyConfig
+}
+
+// ProxyConfig has the proxy configuration
+type ProxyConfig struct {
+	// Top-level basepath for the proxy config
+	Basepath string `yaml:"basepath,omitempty" json:"basepath,omitempty"`
+
+	// A list of Operations, names of which must be unique within the proxy config.
+	Operations []Operation `yaml:"operations,omitempty" json:"operations,omitempty"`
+}
+
+// An API Operation associates a set of rules with a set of request matching
+// settings.
+type Operation struct {
```

Comment:
Will this possibly collide with the API Product Operation definition?

Reply:
They are in different packages so technically no. But I have changed it to `APIOperation`.